influxql/ast.go

```diff
@@ -914,6 +918,28 @@ func (s *SelectStatement) validate(tr targetRequirement) error {
 }
 
 func (s *SelectStatement) validateAggregates(tr targetRequirement) error {
+	// First, determine if specific calls have at least one and only one argument
+	for _, f := range s.Fields {
+		if c, ok := f.Expr.(*Call); ok {
+			switch c.Name {
+			case "derivative", "non_negative_derivative":
+				if exp, got := 1, len(c.Args); got < exp {
```

Comment:
These can take 2 arguments, so it's checking that it got at least one.  Looking at it now it probably should read:

```go
if min, max, got := 1, 2, len(c.Args); got > max || got < min {
```